In [21]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


In [22]:
df = pd.read_csv('house_price.csv')
X = df.iloc[:, :-1].values

            
        #The X variable will contain all rows and all columns except the last one (which is assumed to be the target variable)
     

y = df.iloc[:, -1].values

            #  the y variable will contain the values from the last column (target variable).
            
    
#df

In [23]:
#!pip install scikit-learn
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)


In [32]:
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)


LinearRegression()

In [33]:
print(regressor.coef_)
print(regressor.intercept_)

[  125.85 -1596.93]
90561.37024100643


This means that the base price of house is 90561.37 and the amount will be added/subracted for every 1 sq. units purchased. For some unit increase in area, the price will be increased by 125.853 For some unit increase in no. of rooms, the price will decrease by 1596.93


In [34]:
y_pred = regressor.predict(X_test)
np.set_printoptions(precision=2)
print(np.concatenate((y_pred.reshape(len(y_pred),1), y_test.reshape(len(y_test),1)),1))

[[362648.19 475000.  ]
 [480572.88 579900.  ]
 [318812.02 349900.  ]
 [461734.03 539900.  ]
 [412525.32 499998.  ]
 [337477.5  347000.  ]
 [407239.48 314000.  ]
 [214817.9  169900.  ]
 [287639.53 242900.  ]
 [328329.36 239999.  ]]


In [35]:
from sklearn.metrics import r2_score
r2_score(y_test, y_pred)

0.6745515249932021

### r2 score is bad should be near 1 i.e minimum variability

In [36]:
df.head()

area  rooms   price
0  2104      3  399900
1  1600      3  329900
2  2400      3  369000
3  1416      2  232000
4  3000      4  539900